test/module/irohad/subscription/subscription_test.cpp

```diff
@@ -0,0 +1,655 @@
+ */
+TEST_F(SubscriptionTest, PoolSimpleExecutionTest) {
+  auto manager = createSubscriptionManager<1>();
+  utils::WaitForSingleObject complete;
+
+  std::string test_value = "the fast and the furious";
+  auto subscriber =
+      createSubscriber<std::numeric_limits<uint32_t>::max(), 1ull, std::string>(
+          manager, false, [&complete, test_value](auto, std::string value) {
+            ASSERT_EQ(test_value, value);
+            complete.set();
+          });
+
+  manager->notify(1ull, test_value);
+  ASSERT_TRUE(complete.wait(std::chrono::minutes(1ull)));
+
+  manager->dispose();
+}
+
+/**
+ * @given subscription engine with busy pool
+ * @when subscriber is present in pool threads
+ * @and notification is called
+ * @then subscriber must receive correct data from notification
+ */
+TEST_F(SubscriptionTest, BusyPoolSimpleExecutionTest) {
+  auto manager = createSubscriptionManager<1>();
+
+  utils::WaitForSingleObject complete;
+  utils::WaitForSingleObject complete1;
+  auto subscriber1 =
+      createSubscriber<std::numeric_limits<uint32_t>::max(), 1ull, bool>(
+          manager, false, [&complete, &complete1](auto, auto) {
+            complete1.set();
+            complete.wait();
+            complete.set();
+          });
+
+  std::string test_value = "the fast and the furious";
+  auto subscriber2 =
+      createSubscriber<std::numeric_limits<uint32_t>::max(), 2ull, std::string>(
+          manager, false, [&complete, test_value](auto, std::string value) {
+            ASSERT_EQ(test_value, value);
+            complete.set();
+          });
+
+  manager->notify(1ull, false);
+  complete1.wait();
+  manager->notify(2ull, test_value);
+
+  ASSERT_TRUE(complete.wait(std::chrono::minutes(1ull)));
+  complete.set();
+  manager->dispose();
+}
+
+/**
+ * @given subscription engine
+ * @when 2 subscribers are present
+ * @and they subscribe on a single event in a single thread
+ * @then they must be called in the same order
+ */
+TEST_F(SubscriptionTest, DoubleExecutionTest) {
+  auto manager = createSubscriptionManager<1>();
+  utils::WaitForSingleObject complete;
+
+  uint32_t counter = 0ul;
+  std::string test_value = "the fast and the furious";
+  auto subscriber_1 = createSubscriber<0ul, 1ull, std::string>(
+      manager, false, [&counter, test_value](auto, std::string value) {
+        ASSERT_EQ(test_value, value);
+        ASSERT_EQ(counter, 0ul);
+        ++counter;
+      });
+
+  auto subscriber_2 = createSubscriber<0ul, 1ull, std::string>(
+      manager,
+      false,
+      [&complete, &counter, test_value](auto, std::string value) {
+        ASSERT_EQ(test_value, value);
+        ASSERT_EQ(counter, 1ul);
+        complete.set();
+      });
+
+  manager->notify(1ull, test_value);
+  ASSERT_TRUE(complete.wait(std::chrono::minutes(1ull)));
+
+  manager->dispose();
+}
+
+/**
+ * @given subscription engine
+ * @when 2 subscribers are present
+ * @and they subscribe on a different events in a single thread
```

Comment:
```suggestion
 * @and they subscribe to different events in a single thread
```
Please check similar cases in other test descriptions